### PR TITLE
Azjol nerub a little fix

### PR DESCRIPTION
**Pull request by Nispodzianka**
azjol with this fix
1st boss: https://youtu.be/hLGskpfwB7k?t=122
2nd boss: https://youtu.be/hLGskpfwB7k?t=422

### CODE REVIEW

File: DBM-Party-WotLK/AzjolNerub/Hadronox.lua [hunk 1]
```diff
@@ -7,16 +7,22 @@ mod:SetCreatureID(28921)
 mod:RegisterCombat("combat")
 
 mod:RegisterEventsInCombat(
-	"SPELL_CAST_START 53030 59417 53400 59419"
+	"SPELL_AURA_APPLIED 53030 59417",
+	"SPELL_PERIODIC_DAMAGE 53400 59419"
 )
 
-local warningCloud	= mod:NewSpellAnnounce(53400, 3)
+local specWarningCloud	= mod:NewSpecialWarningMove(53400, nil, nil, nil, 1, 2)
```
Comment (Zidras) [inline DBM-Party-WotLK/AzjolNerub/Hadronox.lua:14]: `NewSpecialWarningGTFO` seems more adequate for this

File: DBM-Party-WotLK/AzjolNerub/Hadronox.lua [hunk 2]
```diff
@@ -7,16 +7,22 @@ mod:SetCreatureID(28921)
 mod:RegisterCombat("combat")
 
 mod:RegisterEventsInCombat(
-	"SPELL_CAST_START 53030 59417 53400 59419"
+	"SPELL_AURA_APPLIED 53030 59417",
+	"SPELL_PERIODIC_DAMAGE 53400 59419"
```
Comment (Zidras) [inline DBM-Party-WotLK/AzjolNerub/Hadronox.lua:11]: missing `"SPELL_PERIODIC_MISSED 53400 59419"`

File: DBM-Party-WotLK/AzjolNerub/Hadronox.lua [hunk 3]
```diff
@@ -7,16 +7,22 @@ mod:SetCreatureID(28921)
...
-end
\ No newline at end of file
+end
+
+function mod:SPELL_PERIODIC_DAMAGE(_, _, _, destGUID, _, _, spellId)
+	if (spellId == 53400 or spellId == 59419) and destGUID == UnitGUID("player") and self:AntiSpam(2, 1) and not self:IsTrivial() then
+		specWarningCloud:Show()
+		specWarningCloud:Play("runaway")
+	end
+end
```
Comment (Zidras) [inline DBM-Party-WotLK/AzjolNerub/Hadronox.lua:28]: missing `mod.SPELL_PERIODIC_MISSED = mod.SPELL_PERIODIC_DAMAGE`

File: DBM-Party-WotLK/AzjolNerub/Krikthir.lua [hunk 4]
```diff
@@ -12,11 +12,13 @@ mod:RegisterEventsInCombat(
 
 local warningCurse	= mod:NewSpellAnnounce(52592, 2)
 
-local timerCurseCD	= mod:NewCDTimer(20, 52592, nil, nil, nil, 2)
+local timerCurseCD	= mod:NewCDTimer(10, 52592, nil, nil, nil, 2)
 
 function mod:SPELL_AURA_APPLIED(args)
 	if args:IsSpellID(52592, 59368) then
-		warningCurse:Show()
+		if args:IsPlayer() then
```
Comment (Zidras) [inline DBM-Party-WotLK/AzjolNerub/Krikthir.lua:19]: warning should pick up everything (maybe disabled by default, but it shouldn't be restricted to player only).
if only player matter, perhaps a new specWarn is better?

**Review by Zidras — CHANGES_REQUESTED** [review]

File: DBM-Party-WotLK/AzjolNerub/Krikthir.lua (hunk 4 above)
Comment (Nispodzianka) [reply DBM-Party-WotLK/AzjolNerub/Krikthir.lua:19]: without `if args:IsPlayer() then` we see warningCurse:Show() everytime any person in PT get this debuff soo if 4 ppl get curse we have spam and we see 3x warningCurse - I dont think this debuff its so important that we need this spam

Comment (Zidras) [reply DBM-Party-WotLK/AzjolNerub/Krikthir.lua:19]: Usually, DBM mod follows this logic:
specWarn for player
warn for others

I don't agree with removing options. My suggestion is:
```lua
		if args:IsPlayer() then
		  specWarnCurse:Show()
	    else
      warningCurse:Show()
end      
```
and warnCurse disabled in the mod default option

**Review by Zidras — APPROVED** [review]

**Pull request merged**